op-chain-ops/squash/sim.go

```diff
@@ -0,0 +1,199 @@
+package squash
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/state"
+)
+
+type staticChain struct {
+	startTime uint64
+	blockTime uint64
+}
+
+func (d *staticChain) Engine() consensus.Engine {
+	return ethash.NewFullFaker()
+}
+
+func (d *staticChain) GetHeader(h common.Hash, n uint64) *types.Header {
+	parentHash := common.Hash{0: 0xff}
+	binary.BigEndian.PutUint64(parentHash[1:], n-1)
+	return &types.Header{
+		ParentHash:      parentHash,
+		UncleHash:       types.EmptyUncleHash,
+		Coinbase:        common.Address{},
+		Root:            common.Hash{},
+		TxHash:          types.EmptyTxsHash,
+		ReceiptHash:     types.EmptyReceiptsHash,
+		Bloom:           types.Bloom{},
+		Difficulty:      big.NewInt(0),
+		Number:          new(big.Int).SetUint64(n),
+		GasLimit:        30_000_000,
+		GasUsed:         0,
+		Time:            d.startTime + n*d.blockTime,
+		Extra:           nil,
+		MixDigest:       common.Hash{},
+		Nonce:           types.BlockNonce{},
+		BaseFee:         big.NewInt(7),
+		WithdrawalsHash: &types.EmptyWithdrawalsHash,
+	}
+}
+
+type simState struct {
+	*state.MemoryStateDB
+	snapshotIndex int
+
+	tempAccessList map[common.Address]map[common.Hash]struct{}
+}
+
+func (db *simState) AddressInAccessList(addr common.Address) bool {
```

Comment:
What interface does this implement? Would like a static interface assertion to ensure changes to upstream geth don't break this

Reply:
These methods could go on the mem state db but also fine with this